datajoint/base_relation.py

```diff
@@ -389,6 +404,8 @@ def drop(self):
         Drop the table and all tables that reference it, recursively.
         User is prompted for confirmation if config['safemode'] is set to True.
         """
+        if self.restriction:
+            raise DataJointError('A relation with an applied restriction condition cannot be dropped.')
```

Comment:
Append how to solve this in the message. Something like `Call drop() directly on the unrestricted UserRelation`

Reply:
agreed.